### PR TITLE
Remove Dockerfile.publish and rewrite existing one during deploy.

### LINKED ISSUE #1204: gcc missing in  openwhisk/dockerskeleton
## Environment details:

* local deployment, vagrant, native ubuntu, Mac OS, Bluemix, ...   
* version of docker, vagrant, ubuntu, ...

## Steps to reproduce the issue:

1.   `./buildAndPush.sh my_namespace/skeleton`

## Provide the expected results and outputs:

```
...
latest: digest: sha256:...
...
```

## Provide the actual results and outputs:

```
$ ./buildAndPush.sh iainh/skeleton
Using iainh/skeleton as the image name
Sending build context to Docker daemon 6.656 kB
Step 1 : FROM openwhisk/dockerskeleton
latest: Pulling from openwhisk/dockerskeleton
e110a4a17941: Pull complete 
4b8a645b1f9e: Pull complete 
30921c0ad717: Pull complete 
bddbfbdffbff: Pull complete 
de0e3cfef544: Pull complete 
e1cca3827cc8: Pull complete 
6dd273025a52: Pull complete 
a6fd5c79e05c: Pull complete 
7efedd5ac033: Pull complete 
Digest: sha256:7670e717d00a1713146ce88a006b4b760e6a3aebf11e6fe9369eeeb383bddd3e
Status: Downloaded newer image for openwhisk/dockerskeleton:latest
 ---> c669352ac080
Step 2 : ENV DEBIAN_FRONTEND noninteractive
 ---> Running in 3d3a8170267e
 ---> 5b3e25422290
Removing intermediate container 3d3a8170267e
Step 3 : ENV FLASK_PROXY_PORT 8080
 ---> Running in 3165dfab51e6
 ---> e211c62b8e36
Removing intermediate container 3165dfab51e6
Step 4 : ADD example.c /action/example.c
 ---> 1013122d9d10
Removing intermediate container e2af339fa232
Step 5 : RUN cd /action; gcc -o exec example.c
 ---> Running in 65d07d734a90
/bin/sh: gcc: not found
The command '/bin/sh -c cd /action; gcc -o exec example.c' returned a non-zero code: 127
Docker build failed
```

### PR DESCRIPTION
**Pull request by rabbah**
Use ansible task to rewrite base image rather than have a second Dockerfile that will get out of sync.
Add test to confirm base image name and inclusion of gcc.

Fixes #1204.

### CODE REVIEW

File: tests/src/system/basic/WskSdkTests.scala [hunk 1]
```diff
@@ -42,18 +45,29 @@ class WskSdkTests
...
+                stdout should include("The docker skeleton is now installed at the current directory.")
+
+            val sdk = new File(dir, "dockerSkeleton")
+            sdk.exists() should be(true)
+            sdk.isDirectory() should be(true)
+
+            val dockerfile = new File(sdk, "Dockerfile")
+            dockerfile.exists() should be(true)
+            dockerfile.isFile() should be(true)
+            val lines = FileUtils.readLines(dockerfile)
```
Comment (markusthoemmes) [inline tests/src/system/basic/WskSdkTests.scala]: Any particular reason why you use Apache Commons vs. `Source.fromPath("myfile.txt").getLines()`?

Comment (rabbah) [inline tests/src/system/basic/WskSdkTests.scala]: Habit.

**Comment by rabbah** [comment]
pg1-378 approved.

**Comment by csantanapr** [comment]
rebase and merged 78ff623d4fa95e35d751e189bb6917f284d7abf5

**Pull request closed without merging**